backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_OrderLine.java

```diff
@@ -57,8 +53,14 @@ public void beforeDelete(@NonNull final I_C_OrderLine orderLine)
 
 		if (contractLogDAO.hasAnyModularLogs(recordReference))
 		{
-			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_ERR_DELETION_NOT_ALLOWED))
+			throw new AdempiereException(MSG_ERR_DELETION_NOT_ALLOWED)
 					.markAsUserValidationError();
 		}
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },ifColumnsChanged = { I_C_OrderLine.COLUMNNAME_M_Product_ID })
+	public void updatePurchaseModularContractId(@NonNull final I_C_OrderLine orderLine)
+	{
+		contractService.updatePurchaseModularContractId(orderLine, false);
```

Comment:
As mentioned above, we must make sure we don't affect non-modular order lines. We can do it either here or in the method ``de.metas.contracts.modular.ModularContractService#updatePurchaseModularContractId`` but we should do it somewhere